Kernel/Syscalls/mmap.cpp

```diff
@@ -595,14 +596,21 @@ KResultOr<FlatPtr> Process::sys$allocate_tls(size_t size)
     if (!range.has_value())
         return ENOMEM;
 
-    auto region_or_error = space().allocate_region(range.value(), String(), PROT_READ | PROT_WRITE);
+    auto region_or_error = space().allocate_region(range.value(), String("Master TLS"), PROT_READ | PROT_WRITE);
     if (region_or_error.is_error())
         return region_or_error.error().error();
 
     m_master_tls_region = region_or_error.value()->make_weak_ptr();
     m_master_tls_size = size;
     m_master_tls_alignment = PAGE_SIZE;
 
+    {
+        Kernel::SmapDisabler disabler;
+        void* fault_at;
+        if (!Kernel::safe_memcpy((char*)m_master_tls_region.unsafe_ptr()->vaddr().as_ptr(), (char*)initial_data.ptr(), size, fault_at))
```

Comment:
I think `copy_from_user` would be safer here (which would also validate the user mode pointer). That implicitly combines `SmapDisabler` with `safe_memcpy`.

Reply:
Please correct me I'm confusing this, but I don't think `copy_from_user` would work here because `m_master_tls_region` is a user address, and `copy_from_user` [doesn't allow that](https://github.com/SerenityOS/serenity/blob/b0faf2287a016a1d1568313ab28a494c4b20398e/Kernel/StdLib.cpp#L195).

Reply:
Thinking about it though there's not reason for `m_master_tls_region` to be user-accessible, so we could change that and then use `copy_from_user`

Reply:
@itamar8910 I think you're right, I didn't realize the destination was also a user accessible pointer...

Reply:
I think making `m_master_tls_region`  non user-accessible can be done in a different PR, so i'm marking this discussion as reolved :)